framework/di/Container.php

```diff
@@ -524,7 +525,7 @@ public function resolveCallableDependencies(callable $callback, $params = [])
                     unset($params[$name]);
                 } elseif (!$associative && isset($params[0]) && $params[0] instanceof $className) {
                     $args[] = array_shift($params);
-                } elseif (Yii::$app->has($name) && ($obj = Yii::$app->get($name)) instanceof $className) {
+                } elseif (Yii::$app instanceof ServiceLocator && Yii::$app->has($name) && ($obj = Yii::$app->get($name)) instanceof $className) {
```

Comment:
Well I've fully removed the check but now the tests fail ;-)

What check would you prefer @klimov-paul ?
Shall I add an isset check?